### ISSUE TITLE
[prototype] The new Transforms API should support different `fill` values for Segmentation masks than Images

### ISSUE THREAD

**Issue by datumbox** [pytorch/vision#6568]
### 🚀 The feature

While migrating the reference scripts for Segmentation to the latest API, @pmeier noticed that one of the transforms uses a different fill value for Images vs Segmentation masks:
https://github.com/pytorch/vision/blob/d367a01a18a3ae6bee13d8be3b63fd6a581ea46f/references/segmentation/transforms.py#L59-L60

The `255` value is used to encode values that the loss should ignore:
https://github.com/pytorch/vision/blob/d367a01a18a3ae6bee13d8be3b63fd6a581ea46f/references/segmentation/train.py#L39

This means that we should have different `fill` values for Images and Segmentation masks. Other libraries such as fvcore [already do this](https://github.com/facebookresearch/fvcore/blob/8cf4acc89e765b263e2afd8a5bcefa8fd677c5f3/fvcore/transforms/transform.py#L756-L757) by adding separate fill `pad_value` and `seg_pad_value` parameters.

It's worth noting that a parameter behaving similarly is `interpolation` only that in the case of masks it's fixed to nearest and thus it's not necessary to have 2 of them.

### Motivation, pitch

Here is what needs to be done:
- [ ] All eligible kernels (`affine`, `rotate`, `pad`, `perspective`, `elastic`, etc) that could receive a `fill` parameter should receive it for for `SegmentationMask`
- [ ] Make it possible for all related Transforms (`Pad`, `RandomZoomOut`, `RandomRotation`, `RandomAffine`, `ElasticTransform`, `FixedSizeCrop`, etc) to receive different parameters for Image and Masks.
- [ ] Update the tests.

cc @vfdev-5 @bjuncek @pmeier

**Comment by vfdev-5** [comment]
Done in https://github.com/pytorch/vision/pull/6599 and https://github.com/pytorch/vision/pull/6586
